bitnami/keycloak/templates/statefulset.yaml

```
@@ -215,19 +215,39 @@ spec:
             - name: KEYCLOAK_EXTRA_ARGS
               value: {{ .Values.extraStartupArgs | quote }}
             {{- end }}
-            {{- if and .Values.adminIngress.enabled .Values.adminIngress.hostname }}
+            {{- if and .Values.adminIngress.enabled .Values.adminIngress.hostname (eq .Values.adminIngress.controller "default") }}
             - name: KC_HOSTNAME_ADMIN_URL
               value: |-
                 {{ ternary "https://" "http://" ( or .Values.adminIngress.tls (eq .Values.proxy "edge") (not (empty .Values.proxyHeaders)) ) -}}
                 {{- include "common.tplvalues.render" (dict "value" .Values.adminIngress.hostname "context" $) -}}
                 {{- include "common.tplvalues.render" (dict "value" .Values.adminIngress.path "context" $) }}
+            {{- else if and .Values.adminIngress.enabled .Values.adminIngress.hostname (eq .Values.adminIngress.controller "gce") }}
+            - name: KC_HOSTNAME_ADMIN_URL
+              value: |-
+                {{ ternary "https://" "http://" ( or .Values.adminIngress.tls (eq .Values.proxy "edge") (not (empty .Values.proxyHeaders)) ) -}}
+                {{- include "common.tplvalues.render" (dict "value" .Values.adminIngress.hostname "context" $) -}}
+                {{- $path := .Values.adminIngress.path -}}
+                {{- if hasSuffix "*" $path -}}
+                  {{- $path = trimSuffix "*" $path -}}
+                {{- end -}}
+                {{- include "common.tplvalues.render" (dict "value" $path "context" $) }}
             {{- end }}
```

I would avoid duplicated lines. What about something like this to reduce the code?

```suggestion
            {{- if and .Values.adminIngress.enabled .Values.adminIngress.hostname }}
            - name: KC_HOSTNAME_ADMIN_URL
              value: |-
                {{ ternary "https://" "http://" ( or .Values.adminIngress.tls (eq .Values.proxy "edge") (not (empty .Values.proxyHeaders)) ) -}}
                {{- include "common.tplvalues.render" (dict "value" .Values.adminIngress.hostname "context" $) -}}
                {{- if eq .Values.adminIngress.controller "default" }}
                  {{- include "common.tplvalues.render" (dict "value" .Values.adminIngress.path "context" $) }}
               {{- else if eq .Values.adminIngress.controller "gce" }}
                 {{- $path := .Values.adminIngress.path -}}
                 {{- if hasSuffix "*" $path -}}
                   {{- $path = trimSuffix "*" $path -}}
                 {{- end -}}
                 {{- include "common.tplvalues.render" (dict "value" $path "context" $) }}
             {{- end }}
           {{- end }}
```

Please double-check correct indentation.

---

Thanks, incorporated